### ISSUE TITLE
[BUG][Ruby Client]Failed to get localized translation when requesting locale(JA, FR...) with upper letters by getString() and translate() interface.

### ISSUE THREAD

**Issue by lyiyu66** [vmware/singleton#1691]
**Describe the bug**
Failed to get localized translation when requesting locale(JA, FR...) with upper letters by getString() and translate() interface.

**To Reproduce**
1. Load config file.
Singleton.load_config(<config file>, <mode>)
2. The translation of en/ja locale:
```
en: about.description: "Use this area to provide additional information"
ja:  "about.description" : "この領域を使用して追加情報を提供する"
fr:  "about.description" : "Utilisez cette zone pour fournir des informations supplémentaires"
```
3. Invoke below interface:
```
expect(SgtnClient::Translation.getString("about", "about.message", "JA")).to eq("この領域を使用して追加情報を提供する")
expect(Singleton.t("about.message", "about", "FR")).to eq("there is one cat in the room")
```
4. See error:
```
expected: "この領域を使用して追加情報を提供する"
            got: "Use this area to provide additional information"

expected: "Utilisez cette zone pour fournir des informations supplémentaires"
            got: "Use this area to provide additional information"
```

**Expected behavior**
expect(SgtnClient::Translation.getString("about", "about.message", "JA")).to eq("この領域を使用して追加情報を提供する")
expect(Singleton.t("about.message", "about", "FR")).to eq("there is one cat in the room")

**Comment by lyiyu66** [comment]
It also is reproducible if requested locale is ZH-HANS, ZH-HANS-CN, ZH-CN

**Comment by lyiyu66** [comment]
The issue is fixed with commit 2807ecb7ae860c0f7b9438dddb4b9310a15a06cd
